lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoOutputFile.java

```diff
@@ -25,25 +25,26 @@
 public interface TrinoOutputFile
 {
     /**
-     * Create file exclusively, failing or overwriting if the file already exists. For file systems which do not
-     * support exclusive creation (e.g. S3), this will fallback to createOrOverwrite().
-     *
-     * @throws FileAlreadyExistsException If a file of that name already exists
+     * This method delegates to {@link #create(AggregatedMemoryContext)}.
      */
     default OutputStream create()
             throws IOException
     {
         return create(newSimpleAggregatedMemoryContext());
     }
 
-    default OutputStream createOrOverwrite()
-            throws IOException
-    {
-        return createOrOverwrite(newSimpleAggregatedMemoryContext());
-    }
+    /**
+     * Create file with the specified content, atomically if possible.
+     * The file will be replaced if it already exists.
```

Comment:
Describe the state after a failure.  I believe the answer is anything is possible, exist, not exist, truncate existing, maybe half written?